pkg/soc/soc.go

```diff
@@ -154,7 +154,7 @@ func FromChunk(sch swarm.Chunk) (*Soc, error) {
 	spanBytes := chunkData[cursor : cursor+swarm.SpanSize]
 	cursor += swarm.SpanSize
 
-	ch, err := contentAddressedChunk(chunkData[cursor:], spanBytes)
+	ch, err := cac.NewWithSpan(chunkData[cursor:], spanBytes)
```

Comment:
You could potentially simplify this with removing `spanBytes` and the cursor increment on `L154-155`, then calling `cac.NewWithDataSpan(chunkData[cursor:])`. This way you can unexport `NewWithSpan` since it is only used here (and simplify the tests as a result).

Reply:
True! Thanks. I fixed it now.